### PR TITLE
feat: argent discount

### LINKED ISSUE #821: Argent 1$ domain page
## Description 

Create a special page for [this](https://www.notion.so/lfg-labs/Argent-x-Starknet-ID-Discount-0bfb0faa247e42499bc5e5542622907e) campaign using the `quantum-leap` campaign. (you can reuse `quantumleap.tsx` and change the wording°.

We need to have: 
- Available only to Argent wallets (mobile and extension)
- Domain costs 1$ for 3 months
- Add argent.stark as a referee on the page directly
- Possibility to subscribe (optional) 
-  24/06-02/07 
- No upsell so you can delete it from `quantumleap.tsx`
- Discount needs to be created on-chain you can ask @Th0rgal for sepolia :)

All campaigns infos here https://www.notion.so/lfg-labs/Argent-x-Starknet-ID-Discount-0bfb0faa247e42499bc5e5542622907e

### PR DESCRIPTION
**Pull request by Marchand-Nicolas**
close: #821 

## Summary by CodeRabbit

- **New Features**
  - Introduced a new discount campaign interface for Argent wallet users, enabling them to view and manage discount offers.
  - Added functionality to connect to the Argent wallet, check for discount expiry, and store referral data.

- **Bug Fixes**
  - Corrected initial state of the renewal box and refined pricing logic in the discount registration process.
  - Removed unused `TxConfirmationModal` component to streamline the user interface.

- **Style**
  - Updated spacing in discount module's styles for improved layout and readability.

### CODE REVIEW

**Review by fricoben — CHANGES_REQUESTED** [review]
## E2E Review 
https://www.loom.com/share/2f14541bd74c46e2aa79cc7eca2ac04d

File: utils/discounts/argent.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    price: "599178082191783",
+    desc: "Secure your domain for the next 3 months at a special discounted price. Don't miss out on this exclusive opportunity to buy your domain for only 1$!",
+    title: { desc: "Get your domain", catch: "for 1$" },
+    image: "/quantumleap/quantumLeapAstro.webp",
+  },
+  name: "Argent 1$ domain",
+  image: "/quantumleap/quantumLeapAstro.webp",
+  expiry: 1719874800 * 1000, // timestamp in ms
+  discountMailGroupId: "98859014745490932",
+  sponsor: "argent.stark",
```
Comment (fricoben) [inline utils/discounts/argent.ts]: No you need to add the address behind argent.stark on mainnet here :)

File: pages/argent.tsx [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    const timestamp = currentDate.getTime();
+
+    if (timestamp >= argentDiscount.expiry) {
+      setScreen(0);
+    }
+  }, []);
+
+  const isArgent = ["Argent X", "Argent (mobile)"].includes(
+    connector.connector?.name ?? ""
+  );
```
Comment (fricoben) [inline pages/argent.tsx]: As said in the E2E you need to give feedback saying this campaign is only for Argent when the user has another wallet.

**Review by fricoben — CHANGES_REQUESTED** [review]
## Code review

Some comments

File: pages/argent.tsx [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+              desc={argentDiscount.offer.desc}
+              image={argentDiscount.offer.image ?? argentDiscount.image}
+              setSearchResult={setSearchResult}
+              setScreen={handleSetScreen}
+              expiry={argentDiscount.expiry}
+            />
+          ) : null}
+          {screen === 2 ? (
+            <div className={styles.container}>
+              <RegisterDiscount
```
Comment (fricoben) [inline pages/argent.tsx:93]: In the `RegisterDiscount` here, terms + subscription checkboxes are checked by default when they should both be unchecked by default

**Review by Th0rgal — APPROVED** [review]
lgtm

**Pull request merged**